pkg/logs/input/kubernetes/convertor.go

```diff
@@ -0,0 +1,86 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package kubernetes
+
+import (
+	"bytes"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	iParser "github.com/DataDog/datadog-agent/pkg/logs/parser"
+	"regexp"
+)
+
+// Convertor for converting kubernetes log line to struct Line.
+type Convertor struct {
+	iParser.Convertor
+}
+
+// Convert validates and converts kubernetes log from byte array to struct Line.
+// Kubernetes log lines follow this pattern '<timestamp> <stream> <flag> <content>',
+// see https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/logs/logs.go
+// Example:
+// 2018-09-20T11:54:11.753589172Z stdout F This is my message
+func (c *Convertor) Convert(msg []byte, defaultPrefix iParser.Prefix) *iParser.Line {
+	components := bytes.SplitN(msg, delimiter, numberOfComponents)
+
+	if len(components) < numberOfComponents-1 {
+		// take this msg as partial log splitted by upstream (line generator).
+		return &iParser.Line{
+			Prefix:  defaultPrefix,
+			Content: msg,
+			Size:    len(msg),
+		}
+	}
+	timestamp := string(components[0])
+	status := string(components[1])
+	flag := string(components[2])
+	if !c.validate(timestamp, status, flag) {
+		// take this msg as partial log splitted by upstream (line generator).
+		return &iParser.Line{
+			Prefix:  defaultPrefix,
+			Content: msg,
+			Size:    len(msg),
+		}
+	}
+
+	if len(components) > numberOfComponents-1 {
+		return &iParser.Line{
+			Prefix: iParser.Prefix{
+				Status:    standardStatus(status),
+				Timestamp: timestamp,
+				Flag:      flag,
+			},
+			Content: components[3],
+			Size:    len(components[3]),
+		}
+	}
+	// only header exists.
+	return nil
+}
+
+// 2019-05-29T13:27:27.482052544Z
```

Comment:
Worth moving those regexes on the top of the file.
For the status & flag checks, do we really need a regex though?